engine/runtime/src/main/java/org/enso/interpreter/runtime/EnsoContext.java

```diff
@@ -101,21 +110,19 @@ public EnsoContext(
     this.inReader = new BufferedReader(new InputStreamReader(environment.in()));
     this.threadManager = new ThreadManager(environment);
     this.resourceManager = new ResourceManager(this);
-    this.isInlineCachingDisabled =
-        environment.getOptions().get(RuntimeOptions.DISABLE_INLINE_CACHES_KEY);
-    var isParallelismEnabled =
-        environment.getOptions().get(RuntimeOptions.ENABLE_AUTO_PARALLELISM_KEY);
+    this.isInlineCachingDisabled = getOption(RuntimeOptions.DISABLE_INLINE_CACHES_KEY);
+    var isParallelismEnabled = getOption(RuntimeOptions.ENABLE_AUTO_PARALLELISM_KEY);
     this.isIrCachingDisabled =
-        environment.getOptions().get(RuntimeOptions.DISABLE_IR_CACHES_KEY) || isParallelismEnabled;
-    this.rootIOPermissions = environment.getOptions().get(EnsoLanguage.IO_ENVIRONMENT);
+        getOption(RuntimeOptions.DISABLE_IR_CACHES_KEY) || isParallelismEnabled;
```

Comment:
It can, and it should not. All the option descriptors should be registered _prior_ to the context initialization. See my other comments about option descriptors.

Reply:
The only known place where it returns `null` is `jobParallelism` - the caller handles `null` there. Otherwise `null` is unexpected and would yield a runtime error.